- Converting Authoring Analyzeconversations of 2023-04-15-preview and 2023-04-01 version to Typespec.

amber-ccc opened the pull request:
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

amber-ccc committed:
- initiated the first version of typespec.

amber-ccc committed:
- deleted testing swagger for authoring conversations
- added testing json for authoring conversations 2023-04-15-preview

amber-ccc committed:
- added long running operations

amber-ccc committed:
- added a createOrUpdate operation with 201 response

amber-ccc committed:
- modified the operationId in SuccesfulCopyProjectAuthorization.json
- compiled

amber-ccc committed:
- added SuccessfulGetDeploymentDeleteFromResourcesStatus.json
- compiled

amber-ccc committed:
- modified model EvaluationResults
- compiled

amber-ccc committed:
- fixed Swagger SpellCheck
- compiled

amber-ccc committed:
- adjusted the map of map structure of ConfusionMatrix
- compiled

amber-ccc committed:
- modified the map of map structure of IntentsEvaluationSummary
- compiled

amber-ccc committed:
- add 2023-04-01 version information

amber-ccc committed:
- added testing json for 2023-04-01
- compiled

amber-ccc committed:
- changed all the Azure.Core.Paged to model definition of array and nextLink (String)

amber-ccc committed:
- modified the parameter name of some testing json of 2023-04-01

amber-ccc committed:
- deleted the analyzeconversations-authoring folder under example of 2023-04-01

amber-ccc committed:
- modified the name of models to add prefix ConversationAnalysisAuthoring
- compiled

amber-ccc committed:
- add version 2023-04-15-preview to CustomConversationSummarization of ConversationalAnalysisAuthoringProjectKind
- compiled

amber-ccc committed:
- changed the way of calling Azure.Core.Foundations.Error to definitions/error
- compiled

amber-ccc committed:
- changed the appId type to uuid
- compiled

amber-ccc committed:
- removed maxLength of projectName of ProjecteMetaData
- compiled

amber-ccc committed:
- added version 2023-04-15-preview to storageInputContainerName of ProjectMetaData

amber-ccc committed:
- added Versions 2023_04_15_preview to some properties
- compiled

amber-ccc committed:
- add Versions v2023_04_15_preview to ConversationalAnalysisAuthoringCreateDeploymentOptions.properties.assignedResourceIds
- compiled

amber-ccc committed:
Merge pull request #2 from amber-ccc/main

merge main to my own branch

amber-ccc committed:
- used npx command to fix TypeSpec Validation errors

amber-ccc committed:
- fixed TypeSpec Validation
- compiled

amber-ccc committed:
- made ConversationalAnalysisAuthoringCreateProjectOptions the request payload of the patch CreateProject operation, and made some of the properties optional
- compiled

amber-ccc committed:
Merge pull request #3 from amber-ccc/main

Upgrade/typespec 0.60 (#30511)

amber-ccc committed:
- fixed typespec validation in Language.AnalyzeConversations-authoring
- compiled

amber-ccc committed:
- modified the titles of each testing json
- compiled

amber-ccc committed:
- revert to the `main` version package-lock.json and package.json

amber-ccc committed:
- modified the title of testing json SuccessfulSwapDeployment
- compiled

amber-ccc committed:
Merge pull request #5 from amber-ccc/main

merge main to conversations branch

amber-ccc committed:
- used npx tsv command to fix TypeSpec Validation

allenjzhang commented on specification/cognitiveservices/data-plane/Language/stable/2023-04-01/analyzeconversations-authoring.json:
```diff
@@ -3,3265 +3,3991 @@
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
+          },
+          {
+            "name": "projectName",
```
You should explicitly model and resume these parameter types in TypeSpec.

amber-ccc committed:
- added resource template for ProjectMetaData
- used ResourceAction and LongRunningResourceAction for Resource ProjectMetaData (Operation authorize-copy and copy)
- compiled

amber-ccc committed:
- added child Resource ProjectDeployment
- implemented Resource ProjectDeployment Template for getDeploymentsList
- compiled

amber-ccc committed:
- added CopyProjectJobState as child Resource of ProjectMetaData
- implemented CopyProjectJobState by ResourceRead
- compiled

amber-ccc committed:
- completed all the LongRunningResourceAction for resource ProjectMetadata
- compiled

amber-ccc committed:
- used resource template for resource ProjectDeployment (Child of ProjectMetadata)
- used ResourceRead template for resource DeploymentJobState (Child of ProjectDeployment)
- compiled

amber-ccc committed:
- added operation templates for resource ProjectTrainedModel
- compiled

amber-ccc committed:
- added some JobState as children of ProjectTrainedModel
- used ResourceRead Template for these JobState resource
- compiled

amber-ccc committed:
- added Resource resources and used some of its resource template
- compiled

amber-ccc committed:
- add resource "train/jobs" for TrainingJobState
- used some resource template of TrainingJobState
- compiled

amber-ccc committed:
- added DeploymentDeleteFromResourcesJobState and SwapDeploymentsJobState to fix duplicated DeploymentJobState resource issue
- compiled

amber-ccc committed:
- added two more jobstate ExportProjectJobState and ImportProjectJobState as resource
- used ResourceRead template for thess two resources
- compiled

amber-ccc committed:
- modified the operation name copy to copyProject
- added action("copy") for this
- compiled

amber-ccc committed:
- used npx tsv command to resolve TypeSpec Validation

amber-ccc committed:
- removed the "read" for @visibility of JobState.jobId
- compiled

amber-ccc committed:
- added @actionSeparator("/:") to solve the RemovedPath and AddedPath issue
- compiled

amber-ccc committed:
- reverted protected files to main version

amber-ccc committed:
- removed all the "read" from @visibility in order to remove the readOnly properties in new swagger
- compiled

amber-ccc committed:
- introduced the low-level LongRunning Operation to remove the status monitor in response
- compiled

amber-ccc committed:
- removed all the FIXME of utcDateTime properties

amber-ccc committed:
- used npx tsv command to resolve TypeSpec Validation

amber-ccc committed:
- removed all the prefix of union and make the x-ms-enum same as old version
- compiled

amber-ccc committed:
- gave concrete example to the uri in SuccessfulGetModelEvaluation testing json, trying to fix Swagger ModelValidation
- compiled

amber-ccc committed:
- modified the namespace, adding Conversations at the end
- compiled
